src/Microsoft.CSharp/tests/IndexingTests.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using Xunit;
+
+namespace Microsoft.CSharp.RuntimeBinder.Tests
+{
+    public class IndexingTests
+    {
+        private class AllTheIntegers
+        {
+            [IndexerName("Integers")]
```

Comment:
Ironically, I think the only place I've ever seen this attribute used outside of example code or tests like the above is in MS.CSharp itself.